### PR TITLE
Feature: Update validation functions so they receive actions instead of proposalId

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task ID: [OS-774]()

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-774]: https://aragonassociation.atlassian.net/browse/OS-774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mathewmeconry — DISMISSED** [review]
I would rename `findAction()` to `findActionIndex()` because it doesn't return the action but only the index of the action within the actions array. 
otherwise LGTM

File: modules/client/test/integration/client/methods.test.ts [hunk 1]
```diff
@@ -1628,7 +1624,7 @@ describe("Client", () => {
         expect(plugins[0].current.release.metadata).toMatchObject(
           releaseMetadata,
         );
-        expect(plugins[0].releases.length).toBe(2)
+        expect(plugins[0].releases.length).toBe(2);
```
Comment (heueristik) [inline modules/client/test/integration/client/methods.test.ts:1627]: Is the SDK repo not linting / format checking itself? This should be a task for the future - not for now.

**Review by mathewmeconry — APPROVED** [review]

File: modules/client/src/internal/client/methods.ts [hunk 2]
```diff
@@ -1110,12 +1109,12 @@ export class ClientMethods extends ClientCore implements IClientMethods {
...
   }
 
-  public isDaoUpdateProposal(
+  public isDaoUpdate(
     actions: DaoAction[],
   ): boolean {
     const initializeFromInterface = DAO__factory.createInterface()
       .getFunction("initializeFrom").format("minimal");
-    return findAction(actions, initializeFromInterface) !== -1;
+    return findActionIndex(actions, initializeFromInterface) !== -1;
```
Comment (heueristik) [inline modules/client/src/internal/client/methods.ts:1117]: I think I've found something here: There will be cases in which we update the DAO **without reinitializing** and thus without calling `initializeFrom`. This method of checking for DAO updates is therefore not sufficient. Shouldn't we rather look for `upgradeTo` or `upgradeToAndCall` in the actions?
@mathewmeconry @Rekard0

**Review by heueristik — CHANGES_REQUESTED** [review]
I have a concern being unrelated to the changes of this PR. Let's take time to discuss it and maybe open a follow-up task.

File: modules/client/src/internal/client/methods.ts (hunk 2 above)
Comment (Rekard0) [reply modules/client/src/internal/client/methods.ts:1117]: we can address it here: https://aragonassociation.atlassian.net/browse/OS-777

**Review by Rekard0 — APPROVED** [review]

File: modules/client/test/integration/client/methods.test.ts (hunk 1 above)
Comment (josemarinas) [reply modules/client/test/integration/client/methods.test.ts:1627]: Yep, createed a task for that https://aragonassociation.atlassian.net/browse/OS-778

**Review by heueristik — APPROVED** [review]

**Pull request merged**